src/libsystemd/sd-event/sd-event.c

```diff
@@ -3418,7 +3417,7 @@ static int dispatch_exit(sd_event *e) {
                 return 0;
         }
 
-        ref = sd_event_ref(e);
+        _unused_ _cleanup_(sd_event_unrefp) sd_event *ref = sd_event_ref(e);
```

Comment:
I have some vague memory that splitting the declaration and the acquisition of the reference prevented a warning from the compiler. But both gcc and clang compile this without warnings, so it must have been in some other scenario.

Reply:
i had to add the `_unused_` thing to make llvm shut up. Without it it did fail to compile.

i.e. there are two ways to avoid the warning:

1. Have at least two lines referencing the object, i.e. the declaration and the acquisition
2. Using `_unused_`, in which case a single line will suffice.

I guess llvm is kinda dumb: once a second line makes use of something it considers it "used". Weird....